posthog/api/survey.py

```diff
@@ -239,7 +242,49 @@ def update(self, instance: Survey, validated_data):
                 instance.targeting_flag.active = False
             instance.targeting_flag.save()
 
-        return super().update(instance, validated_data)
+        instance = super().update(instance, validated_data)
+        self._add_user_survey_interacted_filters(instance)
+        return instance
+
+    def _add_user_survey_interacted_filters(self, instance: Survey, targeting_flag_filters=None):
+        user_submitted_dismissed_filter = {
+            "groups": [
+                {
+                    "variant": "",
+                    "rollout_percentage": 100,
+                    "properties": [
```

Comment:
quick flyby that this doesn't work like we want -> this is currently adding _additional conditions_ to the existing ones, which are OR'ed together, but what we want is to every existing condition to get these properties in their group.

I.e not an extra group, but an addition to whatever they have in their groups.

---

~~Given now that we'll always have a targeting flag for every survey, its potentially worth also making the targeting flag permanent and changing the UX around this - since there's never a case for it not existing.~~

---

Hmm, actually, thinking a bit more, this is getting very messy. I think what works better here is to have a THIRD (ugh I know) flag that's for internal handling: This will handle any repeated surveys iterations (if need be), or dismiss/response calculations, so we can completely hide it from the UI, and it doesn't break the users current flow, nor confuses them with why we automatically magically add and show these extra conditions. [Just make sure we start the name with `survey-targeting-` so it's not billed separately for flags.

And in our SDK the `getActiveMatchingSurveys` will check for this third flag, AND, any linked / targeting flag if it exists.

Reply:
One case I can think of in the future where we might want configuration options here is if someone wants to show the survey again to users who've dismissed it, but not to those who've responded. Don't think this is necessary or vital right now though.

Doing this will again be much easier on a third hidden flag than on existing ones.